From 68e70a48c6fc30abe36d031f461a44dc6f203afe Mon Sep 17 00:00:00 2001
From: Shadi Ammouri <shadi@marvell.com>
Date: Sun, 13 Mar 2016 13:33:32 +0200
Subject: [PATCH 0935/1240] pd: Set u-boot load address to 0x1000 for emulation
 platforms

This overcomes an issue in palladium, where access to address 0x0 might
sometimes fail.

Change-Id: I4e7f2cd19487935669dbfe3293c225b9f223de39
Signed-off-by: Shadi Ammouri <shadi@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/28194
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Haim Boot <hayim@marvell.com>
---
 include/configs/mvebu-common.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/include/configs/mvebu-common.h b/include/configs/mvebu-common.h
index 80d9fa6..22e2c17 100644
--- a/include/configs/mvebu-common.h
+++ b/include/configs/mvebu-common.h
@@ -43,8 +43,15 @@
 #define CONFIG_SYS_MALLOC_F_LEN		0x5000
 #endif
 #ifndef CONFIG_SYS_TEXT_BASE
+#ifdef CONFIG_PALLADIUM
+/* Overcome issue in emulation where writes to address 0x0 might
+   sometimes fail. */
+#define CONFIG_SYS_TEXT_BASE            0x00001000
+#else
 #define CONFIG_SYS_TEXT_BASE            0x00000000
 #endif
+#endif
+
 #define CONFIG_SYS_SDRAM_BASE           0x00000000
 #define CONFIG_SYS_RESET_ADDRESS        0xffff0000
 #define CONFIG_SYS_MALLOC_BASE          (CONFIG_SYS_TEXT_BASE + (3 << 20))  /* TEXT_BASE + 3M */
-- 
1.9.1

